course/src/main/java/in/testpress/course/fragments/DocumentViewerFragment.kt

```diff
@@ -0,0 +1,139 @@
+import com.github.barteksc.pdfviewer.listener.OnLoadCompleteListener
+import com.github.barteksc.pdfviewer.listener.OnPageChangeListener
+import com.github.barteksc.pdfviewer.listener.OnPageErrorListener
+import com.github.barteksc.pdfviewer.scroll.DefaultScrollHandle
+import kotlinx.android.synthetic.main.layout_document_viewer.*
+import kotlinx.android.synthetic.main.layout_document_viewer.emptyContainer
+import kotlinx.android.synthetic.main.layout_pdf_viewer.pdfView
+import kotlinx.android.synthetic.main.layout_pdf_viewer.progressbar
+import java.io.File
+
+
+class DocumentViewerFragment: BaseContentDetailFragment(), PdfDownloadListener,
+        OnPageChangeListener, OnErrorListener, OnLoadCompleteListener, OnPageErrorListener {
+
+    private var pageNumber = 0
+
+    private lateinit var fileName: String
+
+    private lateinit var pdfDownloadListener: PdfDownloadListener
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(
+            inflater: LayoutInflater,
+            container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.layout_document_viewer, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        PRDownloader.initialize(requireContext())
+        pdfDownloadListener = this
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.full_screen_menu, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.fullScreen -> {
+                navigateToPdfViewerActivity()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun navigateToPdfViewerActivity() {
+        startActivity(Intent(this.activity, PdfViewerActivity::class.java).apply {
+            putExtra("pdfUrl", content.attachment?.attachmentUrl)
+            putExtra("pageNumber", pageNumber)
+            putExtra("password", contentId.toString())
+            putExtra("fileName", fileName)
+        })
+    }
+
+    override fun display() {
+        (activity as ContentActivity).setActionBarTitle(content.attachment?.title)
+        fileName = content.attachment?.title ?: ""
+        content.attachment?.attachmentUrl?.let {
+            PdfDownloadUtil(pdfDownloadListener).downloadPdfFromInternet(it,requireContext(),fileName)
+        }
+    }
+
+    override fun isPdfDownloaded(response: Boolean, file: File?) {
+        if (response) {
+            file?.let { showPdfFromFile(file) }?: showErrorView()
+        } else {
+            showErrorView()
+        }
+    }
```

Comment:
`isPdfDownloaded` means `Is Pdf Downloaded ?` This method should return True or False. but it is doing more than one thing